modules/ballerina-core/src/main/java/org/ballerinalang/model/builder/BLangModelBuilder.java

```diff
@@ -1666,6 +1670,20 @@ public void resolvePackageFromNameReference(NameReference nameReference) {
         nameReference.setPkgPath(importPkg.getPath());
     }
 
+    private String validateAndGetPackagePathForServiceProtocol(NodeLocation location, String protocolPkgName) {
```

Comment:
There are two methods already, but non of them can be used here.
Both use NameReference object to do the valiation
method 1 validates the package path and check for undefined import packages as well. If validation fails, it prints a error message saying "packageName:name" (name is null in this case, hence can't use that method)
method 2 just set the package path if exist, it doesn't validate for undefined packages.(we need to check for undefined packages in this case, hence can't use that method as well)